src/main/java/evaluation/optimisation/NTBEA.java

```diff
@@ -233,6 +195,102 @@ public Pair<Object, int[]> run() {
+     * This ensures all metrics from these clones are incorporated into this object's data
+     * @param clones the list of clones that have run in separate threads
+     */
+    protected void collectCloneResults(List<NTBEA> clones) {
+        for (NTBEA clone : clones) {
+            this.elites.addAll(clone.elites);
+            this.winnersPerRun.addAll(clone.winnersPerRun);
+            this.winnerSettings.addAll(clone.winnerSettings);
+            if (clone.bestResult.a.a > this.bestResult.a.a) {
+                bestResult = clone.bestResult;
+            }
+        }
+    }
+
+    protected void activateTournament() {
+        if (!elites.isEmpty()) {
+            // first of all we add the elites into winnerSettings, and winnersPerRun
+            // i.e. we effectively add an extra 'run' for each elite
+            for (int[] elite : elites) {
+                winnerSettings.add(elite);
+                winnersPerRun.add(params.searchSpace.getAgent(elite));
+            }
+        }
+
+        List<AbstractPlayer> players = winnersPerRun.stream().map(p -> (AbstractPlayer) p).collect(Collectors.toList());
+        for (int i = 0; i < players.size(); i++) {
+            players.get(i).setName("Winner " + i + " : " + Arrays.toString(winnerSettings.get(i)));
+        }
+        // Given we have N players in each game, and a total of M agents (the number of NTBEA iterations), we
+        // can reduce the variance in the results (and hence the accuracy of picking the best agent) by using the exhaustive mode
+        // this does rely on not having, say 20 NTBEA iterations on a 6-player game (38k combinations); but assuming
+        // the advice of 10 or fewer iterations holds, then even on a 5-player game we have 252 combinations, which is fine.
+        //double combinationsOfPlayers = CombinatoricsUtils.binomialCoefficientDouble(players.size(), nPlayers);
+        int nTeams = params.byTeam ? game.createGameInstance(nPlayers, params.gameParams).getGameState().getNTeams() : nPlayers;
+        if (players.size() < nTeams) {
+            System.out.println("Not enough players to run a tournament with " + nTeams + " players. Skipping the final tournament - " +
+                               "check the repeats options is at least equal to the number of players.");
+        } else {
+            Map<RunArg, Object> config = new HashMap<>();
+            config.put(matchups, params.tournamentGames);
+            if (players.size() < nPlayers) {
+                // if we don't have enough players to fill the game, then we will need to use self-play
+                config.put(RunArg.mode, "exhaustiveSP");
+            } else {
+                config.put(RunArg.mode, "exhaustive");
+            }
+            config.put(byTeam, true);
+            config.put(RunArg.distinctRandomSeeds, 0);
+            config.put(RunArg.budget, params.budget);
+            config.put(RunArg.verbose, params.verbose);
```

Comment:
I've changed this to pass on the `verbose` run parameter. It seems to me this is intended behaviour, however if this is undesired I can change it back.

Reply:
verbose should not be passed on to the final tournament (verbosity for NTBEA is not the same as verbosity for RoundRobinTournament)